addons/portal/models/portal_mixin.py

```diff
@@ -28,7 +28,7 @@ def _compute_access_url(self):
 
     def _portal_ensure_token(self):
         """ Get the current record access token """
-        if not self.access_token:
+        if not self.sudo().access_token:
             # we use a `write` to force the cache clearing otherwise `return self.access_token` will return False
             self.sudo().write({'access_token': str(uuid.uuid4())})
```

Comment:
Not related to this PR but isn't this comment and this way of doing it obsolete since a long time (many ORM cache improvement PR since then)?
```suggestion
            self.sudo().access_token = str(uuid.uuid4())
```